include/swupd.h

```diff
@@ -64,6 +64,7 @@ struct version_container {
 
 struct header;
 
+extern bool badcert;
```

Comment:
This isn't meant to indicate a bad signature though; if you look at initialize_signature, all it does is load the cert and check for any revocations in the cert chain. Thus a failure is just an indication of a "badcert".  The follow-on logic, and the way I use this flag, is to indicate that when we have a bad cert, do not bother with a signature verification